src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -323,7 +325,25 @@ public SamReader open(final SamInputResource resource) {
                     return reader;
                   }
                 }
-                if (type == InputResource.Type.SEEKABLE_STREAM || type == InputResource.Type.URL) {
+                if (type == InputResource.Type.URI) {
```

Comment:
Some of the comments here are might be irrelevant if we change how we do the input type like I mentioned above.